anchor/anchorparser.go

```diff
@@ -2,45 +2,47 @@ package anchor
 
 import (
 	"crypto/x509"
+	"crypto/x509/pkix"
 	"encoding/asn1"
+	"errors"
+	"fmt"
 	"log"
 
-	"github.com/getyoti/yoti-go-sdk/yotiprotoattr"
-	"github.com/getyoti/yoti-go-sdk/yotiprotocom"
+	"github.com/getyoti/yoti-go-sdk/v2/yotiprotoattr"
+	"github.com/getyoti/yoti-go-sdk/v2/yotiprotocom"
 	"github.com/golang/protobuf/proto"
 )
 
 type anchorExtension struct {
 	Extension string `asn1:"tag:0,utf8"`
 }
 
-var sourceOID = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 47127, 1, 1, 1}
-var verifierOID = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 47127, 1, 1, 2}
+var (
+	sourceOID   = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 47127, 1, 1, 1}
+	verifierOID = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 47127, 1, 1, 2}
+)
 
-//ParseAnchors takes a slice of protobuf anchors, parses them, and returns a slice of Yoti SDK Anchors
+// ParseAnchors takes a slice of protobuf anchors, parses them, and returns a slice of Yoti SDK Anchors
 func ParseAnchors(protoAnchors []*yotiprotoattr.Anchor) []*Anchor {
 	var processedAnchors []*Anchor
 	for _, protoAnchor := range protoAnchors {
 		var extensions []string
-		var anchorType = AnchorTypeUnknown
-
-		var parsedCerts = parseCertificates(protoAnchor.OriginServerCerts)
-
+		var (
+			anchorType  = AnchorTypeUnknown
+			parsedCerts = parseCertificates(protoAnchor.OriginServerCerts)
+		)
 		for _, cert := range parsedCerts {
 			for _, ext := range cert.Extensions {
-				var ae anchorExtension
-				if sourceOID.Equal(ext.Id) {
-					anchorType = AnchorTypeSource
-				} else if verifierOID.Equal(ext.Id) {
-					anchorType = AnchorTypeVerifier
-				}
-
-				_, err := asn1.Unmarshal(ext.Value, &ae)
-				if err == nil {
-					extensions = append(extensions, ae.Extension)
-				} else {
-					log.Printf("Error unmarshalling anchor extension: %q", err)
+				var (
+					value string
+					err   error
+				)
+				anchorType, value, err = parseExtension(ext)
+				if err != nil {
+					log.Printf("error parsing anchor extension, %v", err)
```

Comment:
I know I suggested this, but I've realised we will now log an error for other extensions, which is incorrect. Maybe it would be better to change the `default` case in `parseExtension` to be:
```go
default:
	return anchorType, "", nil
```
Then you can do:
```go
if err != nil {
	⋮
} else if anchorType == AnchorTypeUnknown {
	continue
}
⋮
```

Reply:
Yes that's sensible, will change